types/npm-whoami/index.d.ts

```diff
@@ -0,0 +1,24 @@
+// Type definitions for npm-whoami 1.1
+// Project: https://github.com/jamestalmage/npm-whoami#readme
+// Definitions by: Josh Goldberg <https://github.com/JoshuaKGoldberg>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = NpmWhoami;
+
+declare function NpmWhoami(callback: NpmWhoami.NpmWhoamiCallback, opts?: NpmWhoami.NpmWhoamiOptions): void;
+
+declare namespace NpmWhoami {
+    interface NpmWhoamiCallback {
+        (err: Error, username: undefined): void;
+        (err: null, username: string): void;
+    }
+
+    type NpmWhoamiOptions = number | string | NpmWhoamiBothOptions;
+
+    interface NpmWhoamiBothOptions {
+        registry?: string;
+        timeout?: number;
+    }
+
+    function sync(opts: NpmWhoamiOptions): string | undefined;
+}
```

Comment:
```suggestion
export = whoami;

declare function whoami(callback: whoami.Callback, opts?: whoami.Options): void;

declare namespace whoami {
    interface Callback {
        (err: Error, username: undefined): void;
        (err: null, username: string): void;
    }

    type Options = number | string | BothOptions;

    interface BothOptions {
        registry?: string;
        timeout?: number;
    }

    function sync(opts: Options): string | undefined;
}
```
that way original symbol names could be preserved (notice no change in tests required)

Reply:
Thanks, this is great!

Just added `Whoami` before `Callback` and `Options` - I've always found it annoying to have to rename constantly...